### ISSUE TITLE
cups_queue fails when trying to manage an ipp:// queue

### ISSUE THREAD

**Issue by lesinigo** [leoarnold/puppet-cups#28]
**Given** Ubuntu 16.04 LTS with Puppet 4.10.9 and CUPS 2.1.3-4ubuntu0.3 and leoarnold/cups v2.0.1

**When** I apply the manifest

```puppet
include '::cups'
```

with this hiera data

```yaml
cups::default_queue: MyPrinter
cups::web_interface: true
cups::resources:
  MyPrinter:
    ensure: printer
    uri: 'ipp://192.168.10.20/printers/MyPrinter'
    make_and_model: 'Generic PostScript Printer'
```
**In order to** configure a generic PS printer that will send everything to another cups host

**Then** I get the error message:

```Shell
Error: Execution of '/usr/sbin/lpadmin -E -p MyPrinter -o printer-is-shared=false' returned 1: lpadmin: Cannot change printer-is-shared for remote queues.
```

As far as I can tell, the `cups_queue` resource insists on setting the `printer-is-shared` attribute, but [it cannot be set with IPP queues](https://github.com/apple/cups/issues/4766#issuecomment-197693092).

I get the exact same behavior with `shared: true` and `shared: false`.
It seems to me that `cups_queue` should not try to set that parameter at all if the uri starts with `ipp://` or one of its variations (`ipps` and so on...)

**Comment by leoarnold** [comment]
Wow, thank you for digging up the underlying CUPS issue! I'll just remove that default value.

Nevertheless, if you want to create a `Generic PostScript Printer`, you'll still have to add `model: drv:///sample.drv/generic.ppd`.

**Comment by leoarnold** [comment]
This issue is fixed in release 2.0.2, available now in the Forge and as a GitHub release.

**Comment by lesinigo** [comment]
Tested and working perfectly in the use case of this issue using release 2.0.2 from the forge.
Thank you so much for the quick response!
